### PR TITLE
include connection and store path in dataspace analytics result

### PR DESCRIPTION
**Pull request by YannanGao-gs**
#### What type of PR is this?

Improvement

#### What does this PR do / why is it needed ?

include connection and store path in dataspace analytics result

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/model/DataSpaceExecutionContextAnalysisResult.java [hunk 1]
```diff
@@ -27,6 +27,7 @@ public class DataSpaceExecutionContextAnalysisResult
     public String mapping;
     public List<String> compatibleRuntimes;
     public String defaultRuntime;
+    public DataSpaceExecutionContextMetadata metadata;
```
Comment (MauricioUyaguari) [inline legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/model/DataSpaceExecutionContextAnalysisResult.java:30]: should we rename this to `runtimeMetadata` ?

File: legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/model/DataSpaceExecutionContextMetadata.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.generation.analytics.model;
+
+public class DataSpaceExecutionContextMetadata
```
Comment (MauricioUyaguari) [inline legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/model/DataSpaceExecutionContextMetadata.java:17]: `DataSpaceExecutionContextRuntimeMetadata` ?

File: legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java [hunk 3]
```diff
@@ -314,9 +339,30 @@ public static DataSpaceAnalysisResult analyzeDataSpace(Root_meta_pure_metamodel_
...
             excResult.defaultRuntime = HelperModelBuilder.getElementFullPath(executionContext._defaultRuntime(), pureModel.getExecutionSupport());
+            Optional<org.finos.legend.engine.protocol.pure.v1.model.packageableElement.PackageableElement> packageableRuntime = pureModelContextData.getElements().stream().filter(e -> e.getPath().equals(excResult.defaultRuntime) && e instanceof PackageableRuntime).findFirst();
+            if (packageableRuntime.isPresent() && packageableRuntime.get() instanceof PackageableRuntime)
+            {
+                PackageableRuntime runtime = (PackageableRuntime) packageableRuntime.get();
+                if (runtime.runtimeValue.connections != null && !runtime.runtimeValue.connections.isEmpty() && !runtime.runtimeValue.connections.get(0).storeConnections.isEmpty())
+                {
+                    String storePath = runtime.runtimeValue.connections.get(0).store.path;
+                    Connection connection = runtime.runtimeValue.connections.get(0).storeConnections.get(0).connection;
+                    if (connection instanceof ConnectionPointer)
```
Comment (MauricioUyaguari) [inline legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java:350]: shouldnt we handle embedded connections here  as well ?

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

File: legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java (hunk 3 above)
Comment (YannanGao-gs) [reply legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java:350]: Currently only support what this feature has covered before 
https://github.com/finos/legend-studio/pull/3508/files#diff-0e9be2294b5afd6e1687f59f976f0640a5e79aa8c6d5db88df497bad86319084L98

**Review by MauricioUyaguari — DISMISSED** [review]

File: legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java (hunk 3 above)
Comment (MauricioUyaguari) [reply legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java:350]: can we include it tho. no reason i see it shouldnt be covered.

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**